File: articles/situational-supports-for-sustained-effort/c/situational_support_index.c
#include <stdio.h>

double support_index(
    double autonomy,
    double feedback,
    double belonging,
    double mentoring,
    double recovery,
    double resources,
    double fairness,
    double safety
) {
    return 0.16 * autonomy
         + 0.16 * feedback
         + 0.16 * belonging
         + 0.12 * mentoring
         + 0.14 * recovery
         + 0.14 * resources
         + 0.12 * fairness
         + 0.10 * safety;
}

int main(void) {
    double index = support_index(4.2, 4.3, 4.1, 3.9, 4.0, 3.8, 4.1, 4.2);
    printf("Synthetic situational support index: %.3f\n", index);
    printf("Professional caution: synthetic demonstration only.\n");
    return 0;
}

File: articles/situational-supports-for-sustained-effort/rust/situational_support_index.rs
fn support_index(
    autonomy: f64,
    feedback: f64,
    belonging: f64,
    mentoring: f64,
    recovery: f64,
    resources: f64,
    fairness: f64,
    safety: f64,
) -> f64 {
    0.16 * autonomy
        + 0.16 * feedback
        + 0.16 * belonging
        + 0.12 * mentoring
        + 0.14 * recovery
        + 0.14 * resources
        + 0.12 * fairness
        + 0.10 * safety
}

fn main() {
    let index = support_index(4.2, 4.3, 4.1, 3.9, 4.0, 3.8, 4.1, 4.2);
    println!("Synthetic situational support index: {:.3}", index);
    println!("Professional caution: synthetic demonstration only.");
}
